### PR TITLE
env variables are moved to json config file

### PR DESCRIPTION
**Pull request by ikaratass**
* network interface is still configured in the env file to run the simulator according to secc settings. 
* Energytransfermode could be set
* certificate installation required flag can be configured.
* For each configuration can be found under the example folder.
* The items below have moved to the JSON config file.
     supported_protocols
      supported_energy_services
      is_cert_install_needed
      use_tls
      sdp_retry_cycles
      max_contract_certs
      enforce_tls
      max_supporting_points

### CODE REVIEW

File: iso15118/evcc/evcc_config.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    )
+    is_cert_install_needed: bool = Field(None, alias="isCertInstallNeeded")
+    # Indicates the security level (either TCP (unencrypted) or TLS (encrypted))
+    # the EVCC shall send in the SDP request
+    use_tls: Optional[bool] = Field(None, alias="useTls")
+    sdp_retry_cycles: Optional[int] = Field(None, alias="sdpRetryCycles")
+    max_contract_certs: Optional[int] = Field(None, alias="maxContractCerts")
+    enforce_tls: bool = Field(None, alias="enforceTls")
+    supported_protocols: Optional[List[str]] = Field(
+        None, max_items=4, alias="supportedProtocols"
```
Comment (shalinnijel2) [inline iso15118/evcc/evcc_config.py:25]: We could also potentially have -20 WPT and-20 ACDP. Maybe better to set this to 6.

File: iso15118/shared/utils.py [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+    read_settings = list(set(read_settings))
+    return read_settings
+
+
+def load_requested_protocols(read_protocols: Optional[List[str]]) -> List[Protocol]:
+    supported_protocols = [
+        "ISO_15118_2",
+        "ISO_15118_20_AC",
+        "DIN_SPEC_70121",
+    ]
```
Comment (shalinnijel2) [inline iso15118/shared/utils.py:23]: This could now include "ISO_15118_20_DC"

File: iso15118/shared/utils.py [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
 logger = logging.getLogger(__name__)
 
 
+def _format_list(read_settings: List[str]) -> List[str]:
+    read_settings = list(filter(None, read_settings))
+    read_settings = [setting.strip().upper() for setting in read_settings]
+    read_settings = list(set(read_settings))
+    return read_settings
+
+
```
Comment (shalinnijel2) [inline iso15118/shared/utils.py:17]: As this is in shared, the one in secc_settings.py could perhaps be removed?
https://github.com/SwitchEV/iso15118/blob/2b146b483996863e82fca111db33dabf39f293fc/iso15118/secc/secc_settings.py#L19

File: iso15118/shared/utils.py [hunk 4]
```diff
@@ -1,10 +1,64 @@
...
+    protocols = _format_list(read_protocols)
+    valid_protocols = list(set(protocols).intersection(supported_protocols))
+    if not valid_protocols:
+        raise NoSupportedProtocols(
+            f"No supported protocols configured. Supported protocols are "
+            f"{supported_protocols} and could be configured in evcc_config.json"
+        )
+    supported_protocols = [Protocol[x] for x in valid_protocols]
+    return supported_protocols
```
Comment (shalinnijel2) [inline iso15118/shared/utils.py:33]: Same comment as above. The definition in secc_settings.py could be removed and use this one instead?

https://github.com/SwitchEV/iso15118/blob/2b146b483996863e82fca111db33dabf39f293fc/iso15118/secc/secc_settings.py#L126

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]

File: iso15118/evcc/controller/simulator.py [hunk 5]
```diff
@@ -168,11 +173,14 @@ async def get_energy_transfer_mode(
         """Overrides EVControllerInterface.get_energy_transfer_mode()."""
         if protocol == Protocol.DIN_SPEC_70121:
             return EnergyTransferModeEnum.DC_EXTENDED
-        return EnergyTransferModeEnum.AC_THREE_PHASE_CORE
+        else:
+            return self.config.energy_transfer_mode
```
Comment (shalinnijel2) [inline iso15118/evcc/controller/simulator.py:177]: Do we need the if check now? Couldn't we return self.config.energy_transfer_mode directly?

File: iso15118/evcc/evcc_config.py (hunk 1 above)
Comment (ikaratass) [reply iso15118/evcc/evcc_config.py]: Done

File: iso15118/evcc/evcc_config.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+from iso15118.shared.messages.enums import UINT_16_MAX, EnergyTransferModeEnum
+
+logger = logging.getLogger(__name__)
+
+
+class EVCCConfig(BaseModel):
+    supported_energy_services: List[str] = Field(
+        None, max_items=4, alias="supportedEnergyServices"
+    )
+    is_cert_install_needed: bool = Field(None, alias="isCertInstallNeeded")
```
[the remark this answers is not in the document]
Comment (ikaratass) [reply iso15118/evcc/evcc_config.py:17]: Done

File: iso15118/evcc/controller/simulator.py (hunk 5 above)
Comment (ikaratass) [reply iso15118/evcc/controller/simulator.py:177]: as it is an enumeration of int, better to define a string to enum conversion.

Comment (ikaratass) [reply iso15118/evcc/controller/simulator.py:177]: Done

**Review by shalinnijel2 — APPROVED** [review]

**Pull request merged**